scripts/build/gen_kobject_list.py

```diff
@@ -113,7 +113,8 @@
     ("ztest_unit_test", ("CONFIG_ZTEST_NEW_API", True, False)),
     ("ztest_test_rule", ("CONFIG_ZTEST_NEW_API", True, False)),
     ("rtio", ("CONFIG_RTIO", False, False)),
-    ("rtio_iodev", ("CONFIG_RTIO", False, False))
+    ("rtio_iodev", ("CONFIG_RTIO", False, False)),
+    ("sensor_decoder_api", ("CONFIG_SENSOR_ASYNC_API", True, False))
```

Comment:
Is the decoder struct a kobject then? This seems strange as its a const struct with function pointers. Can you help explain the reasoning behind this?

Reply:
I found this bit in the kObject documentation:

>  - The second item is a boolean indicating whether it is permissible for
>    the object to be located in user-accessible memory.

So this was the easiest way to make sure that the decoders can be seen from anywhere.

Reply:
Ah nice, makes sense.